Doc Sweep AspNetDocs

wadepickett opened the pull request:
[Internal Review: Register User section of: Secure a Web API with Individual Accounts and Local Login in ASP.NET Web API 2.2   ](https://review.docs.microsoft.com/en-us/aspnet/web-api/overview/security/individual-accounts-in-web-api?branch=pr-en-us-513#register-a-user)

[Internal Review: Understanding Database Authentication Options: Protecting Connection Strings and Other Configuration Information (C#)  ](https://review.docs.microsoft.com/en-us/aspnet/web-forms/overview/data-access/advanced-data-access-scenarios/protecting-connection-strings-and-other-configuration-information-cs?branch=pr-en-us-513#understanding-database-authentication-options)

[Internal Review: Using the Web Deploy Handler:  Choosing the Right Approach to Web Deployment](https://review.docs.microsoft.com/en-us/aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/choosing-the-right-approach-to-web-deployment?branch=pr-en-us-513#using-the-web-deploy-handler)

[Internal Review: Choosing the Right Approach to Web Deployment: Introduction to Working with a Database in ASP.NET Web Pages (Razor) Sites](https://review.docs.microsoft.com/en-us/aspnet/web-pages/overview/data/5-working-with-data?branch=pr-en-us-513#deleting-data-in-a-database)

[Internal Review: Adding a Custom Role: Membership and Administration (Wingtip Toys)](https://review.docs.microsoft.com/en-us/aspnet/web-forms/overview/getting-started/getting-started-with-aspnet-45-web-forms/membership-and-administration?branch=pr-en-us-513#adding-a-custom-role)

Fixes #514

Related PR's: #511

For internal review links see build details.  They are 18 documents scattered across the repo.

Rick-Anderson commented on aspnet/web-api/overview/security/individual-accounts-in-web-api/samples/sample5.cmd:
```diff
@@ -7,4 +7,4 @@ X-Requested-With: XMLHttpRequest
 
-{"Email":"alice@example.com","Password":"Password1!","ConfirmPassword":"Password1!"}
\ No newline at end of file
+{"Email":"alice@example.com","Password":$CREDENTIAL_PLACEHOLDER$,"ConfirmPassword":"$CREDENTIAL_PLACEHOLDER$"}
```
```suggestion
{"Email":"alice@example.com",$CREDENTIAL_PLACEHOLDER`$,$CREDENTIAL_PLACEHOLDER2$"}
```

wadepickett replied:
Thanks, yes, I couldn't see how to handle this any better with what's there to work with.  I was trying to avoid spiraling into much more major rework of the doc and samples unless we really needed to.

Rick-Anderson commented on aspnet/web-forms/overview/data-access/advanced-data-access-scenarios/protecting-connection-strings-and-other-configuration-information-cs.md:
```diff
@@ -164,9 +164,9 @@ These tutorials have used Windows authentication. You can tell what authenticati
+The following example shows a connection string that uses SQL authentication, and where `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair. Note the credentials are embedded within the connection string:
 
-`Server=serverName; Database=Northwind; uid=userID; pwd=password`
+`Server=serverName; Database=Northwind; uid=userID; pwd=$CREDENTIAL_PLACEHOLDER$`
```
```suggestion
`Server=serverName; Database=Northwind; uid=userID; $CREDENTIAL_PLACEHOLDER$`
```

Add where ...

Rick-Anderson commented on aspnet/web-forms/overview/deployment/advanced-enterprise-web-deployment/performing-a-what-if-deployment/samples/sample1.cmd:
```diff
@@ -4,7 +4,7 @@ MSDeploy.exe
         computerName="https://stageweb1:8172/MSDeploy.axd?site=DemoSite",
         username="FABRIKAM\stagingdeployer",
-        password="Pa$$w0rd",
+        password=$CREDENTIAL_PLACEHOLDER$,
```
Revert for now. We need to come up with a different approach.

wadepickett replied:
```suggestion
        password="Pa$$w0rd",
```

wadepickett replied:
```suggestion
        password="Pa$$w0rd",
```

Rick-Anderson commented on aspnet/web-forms/overview/deployment/advanced-enterprise-web-deployment/performing-a-what-if-deployment.md:
```diff
@@ -33,7 +33,7 @@ Because a "what if" deployment doesn't actually change anything on the destinati
 As described in [Deploying Web Packages](../web-deployment-in-the-enterprise/deploying-web-packages.md), you can deploy web packages using Web Deploy in two ways&#x2014;by using the MSDeploy.exe command-line utility directly or by running the *.deploy.cmd* file that the build process generates.
 
-If you're using MSDeploy.exe directly, you can run a "what if" deployment by adding the **–whatif** flag to your command. For example, to evaluate what would happen if you deployed the ContactManager.Mvc.zip package to a staging environment, the MSDeploy command should resemble this:
+If you're using MSDeploy.exe directly, you can run a "what if" deployment by adding the **–whatif** flag to your command. For example, to evaluate what would happen if you deployed the ContactManager.Mvc.zip package to a staging environment, the MSDeploy command should resemble this, where `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair.
```
revert until we figure this one out.

wadepickett replied:
```suggestion
If you're using MSDeploy.exe directly, you can run a "what if" deployment by adding the **–whatif** flag to your command. For example, to evaluate what would happen if you deployed the ContactManager.Mvc.zip package to a staging environment, the MSDeploy command should resemble this:
```

Rick-Anderson commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/configuring-deployment-properties-for-a-target-environment/samples/sample2.xml:
```diff
@@ -8,7 +8,7 @@
     </MSDeployUsername>
     <MSDeployPassword Condition=" '$(MSDeployPassword)'=='' ">
-      Pa$$w0rd
+      $CREDENTIAL_PLACEHOLDER$
```
revert

wadepickett replied:
```suggestion
      Pa$$w0rd
```

Rick-Anderson commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/configuring-deployment-properties-for-a-target-environment/samples/sample2.xml:
```diff
@@ -21,7 +21,7 @@
     </ContactManagerServiceIisPath>
     <CmDatabaseConnectionString Condition=" '$(CmDatabaseConnectionString)'=='' ">
-      Data Source=STAGEDB1;User ID=sa;Password=Pa$$w0rd;
+      Data Source=STAGEDB1;User ID=sa;Password=$CREDENTIAL_PLACEHOLDER$;
```
revert

Rick-Anderson commented on aspnet/web-forms/overview/deployment/web-deployment-in-the-enterprise/deploying-web-packages.md:
```diff
@@ -121,7 +121,7 @@ In addition, you'll need to specify various other [provider-specific settings](h
 2. Modify the *ContactManager.Mvc.SetParameters.xml* file to contain the correct parameter values for your staging environment, as described in [Configuring Parameters for Web Package Deployment](configuring-parameters-for-web-package-deployment.md).
 3. Open a Command Prompt window and browse to the location of MSDeploy.exe. This is typically at %PROGRAMFILES%\IIS\Microsoft Web Deploy V2\msdeploy.exe.
-4. Type this command, and then press Enter (disregard the line breaks):
+4. Type this command, where `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair. Then press Enter (disregard the line breaks):
```
revert

wadepickett replied:
```suggestion
4. Type this command, and then press Enter (disregard the line breaks):
```

Rick-Anderson commented on aspnet/web-forms/overview/getting-started/getting-started-with-aspnet-45-web-forms/membership-and-administration/samples/sample3.cs:
```diff
@@ -41,7 +41,7 @@ internal void AddUserAndRole()
         Email = "canEditUser@wingtiptoys.com"
       };
-      IdUserResult = userMgr.Create(appUser, "Pa$$word1");
+      IdUserResult = userMgr.Create(appUser, Configuration["MyPasswordKey"]);
```
```suggestion
      IdUserResult = userMgr.Create(appUser, Configuration["AppUserPasswordKey"]);
```

Rick-Anderson commented on aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-vb/samples/sample5.xml:
```diff
@@ -10,9 +10,9 @@
  validation="SHA1"
 
- decryptionKey="1513F567EE75F7FB5AC0AC4D79E1D9F25430E3E2F1BCDD3370BCFC4EFC97A541"
+ decryptionKey=$CREDENTIAL_PLACEHOLDER$
```
Revert these until we get an approved approach.

wadepickett replied:
```suggestion
 decryptionKey=$CREDENTIAL_PLACEHOLDER$
```

wadepickett replied:
```suggestion
 decryptionKey="1513F567EE75F7FB5AC0AC4D79E1D9F25430E3E2F1BCDD3370BCFC4EFC97A541"
```

Rick-Anderson commented on aspnet/web-pages/overview/data/5-working-with-data/samples/sample27.xml:
```diff
@@ -3,7 +3,7 @@
    <add
      name="SQLServerConnectionString"
-     connectionString= "server=myServer;database=myDatabase;uid=username;pwd=password"
+     connectionString= "server=myServer;database=myDatabase;uid=username;pwd=$CREDENTIAL_PLACEHOLDER$"
```
```suggestion
     connectionString= "server=myServer;database=myDatabase;uid=username;$CREDENTIAL_PLACEHOLDER$"
```

Where $CREDENTIAL_PLACEHOLDER$ is the ... (not key/value pair, but use the correct term)

Rick-Anderson reviewed (changes requested):
* We can never show putting a password in code.
* We can't replace one password with another named `$CREDENTIAL_PLACEHOLDER$`, we need to get another approved approach. That's too close to working code that can be copy/pasted.

wadepickett commented:
I don't have the "Add suggestion to batch" option on this repo, so the commits will be added one at a time.

wadepickett commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/configuring-deployment-properties-for-a-target-environment/samples/sample2.xml:
```diff
@@ -21,10 +21,10 @@
     </ContactManagerServiceIisPath>
     <CmDatabaseConnectionString Condition=" '$(CmDatabaseConnectionString)'=='' ">
-      Data Source=STAGEDB1;User ID=sa;Password=Pa$$w0rd;
+      Data Source=STAGEDB1;User ID=sa;Password=$CREDENTIAL_PLACEHOLDER$;
```
```suggestion
      Data Source=STAGEDB1;User ID=sa;Password=Pa$$w0rd;
```

wadepickett commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/configuring-deployment-properties-for-a-target-environment.md:
```diff
@@ -89,6 +89,7 @@ In this example:
 - You're instructing Web Deploy to use basic authentication.
 - You're specifying that Web Deploy should impersonate the FABRIKAM\stagingdeployer account on the remote computer.
 - You're using SQL Server authentication to deploy the **ContactManager** database to STAGEDB1.
+- Note that `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair.
```
```suggestion
```

wadepickett commented on aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-vb.md:
```diff
@@ -197,7 +197,7 @@ By default, the keys used for encryption and validation are generated automatica
 When working in a web farm setting or sharing authentication tickets across applications on the same server, you will need to configure the &lt;machineKey&gt; element in the affected applications so that their decryptionKey and validationKey values match up.
 
-While neither of the above scenarios applies to our sample application, we can still specify explicit decryptionKey and validationKey values and define the algorithms to be used. Add a &lt;machineKey&gt; setting to the Web.config file:
+While neither of the above scenarios applies to our sample application, we can still specify explicit decryptionKey and validationKey values and define the algorithms to be used. Add a &lt;machineKey&gt; setting to the Web.config file. Note the following example is using `$CREDENTIAL_PLACEHOLDER$` as placeholder for the `decryptionKey` and `validationKey` key-value pairs:
```
```suggestion
While neither of the above scenarios applies to our sample application, we can still specify explicit decryptionKey and validationKey values and define the algorithms to be used. Add a &lt;machineKey&gt; setting to the Web.config file:
```

wadepickett commented:
@Rick-Anderson Reverted where suggested.  ( I made the conversation history a bit long by doing this line by line in the GitHub UI, but that was much faster to fix in the UI here and pull back to local. Ignore my unconventional workflow.)

Rick-Anderson commented on aspnet/web-api/overview/security/individual-accounts-in-web-api/samples/sample5.cmd:
```diff
@@ -7,4 +7,4 @@ X-Requested-With: XMLHttpRequest
 
-{"Email":"alice@example.com","Password":"Password1!","ConfirmPassword":"Password1!"}
\ No newline at end of file
+{"Email":"alice@example.com",$CREDENTIAL_PLACEHOLDER`$,$CREDENTIAL_PLACEHOLDER2$"}
```
```suggestion
{"Email":"alice@example.com",$CREDENTIAL_PLACEHOLDER1$,$CREDENTIAL_PLACEHOLDER2$"}
```

Rick-Anderson commented on aspnet/web-forms/overview/data-access/advanced-data-access-scenarios/protecting-connection-strings-and-other-configuration-information-cs.md:
```diff
@@ -164,9 +164,9 @@ These tutorials have used Windows authentication. You can tell what authenticati
 The Integrated Security=True and lack of a username and password indicate that Windows authentication is being used. In some connection strings the term Trusted Connection=Yes or Integrated Security=SSPI is used instead of Integrated Security=True, but all three indicate the use of Windows authentication.
 
-The following example shows a connection string that uses SQL authentication. Note the credentials embedded within the connection string:
+The following example shows a connection string that uses SQL authentication, and where `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair. Note the credentials are embedded within the connection string:
```
```suggestion
The following example shows a connection string that uses SQL authentication. The `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair. Note the credentials are embedded within the connection string:
```

Rick-Anderson commented on aspnet/web-forms/overview/getting-started/getting-started-with-aspnet-45-web-forms/membership-and-administration/samples/sample3.cs:
```diff
@@ -41,7 +41,7 @@ internal void AddUserAndRole()
         Email = "canEditUser@wingtiptoys.com"
       };
-      IdUserResult = userMgr.Create(appUser, "Pa$$word1");
+      IdUserResult = userMgr.Create(appUser, Configuration["AppUserPasswordKey"]);
```
Sorry, I was thinking this was ASP.NET Core. You'll need to use the Web Forms configuration approach.

Rick-Anderson commented on aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-cs.md:
```diff
@@ -54,11 +54,11 @@ Next, specify the various database connection information (see Figure 2). When y
 **Figure 2**: Specify the Database Connection Information ([Click to view full-size image](configuring-the-production-web-application-to-use-the-production-database-cs/_static/image6.jpg))
 
-The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the  ...
+The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the  ...
```
```suggestion
The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the following,. `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair:
```

Rick-Anderson commented on aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-vb.md:
```diff
@@ -54,11 +54,11 @@ Next, specify the various database connection information (see Figure 2). When y
 **Figure 2**: Specify the Database Connection Information ([Click to view full-size image](configuring-the-production-web-application-to-use-the-production-database-vb/_static/image6.jpg))
 
-The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the  ...
+The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the  ...
```
same fix, break up sentence

wadepickett replied:
```suggestion
The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the following. Note `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair:
```

wadepickett replied:
```suggestion
The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the following. Note `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair:
```

Rick-Anderson commented on aspnet/web-pages/overview/data/5-working-with-data.md:
```diff
@@ -360,7 +360,7 @@ This section shows how to let users delete a product from the *Product* database
 > The second way is to use a connection string. A connection string contains information about how to connect to a database. This can include a file path, or it can include the name of a SQL Server database on a local or remote server, along with a user name and password to connect to that server. (If you keep data in a centrally managed version of SQL Server, such as on a hosting provider's site ...
 > 
-> In WebMatrix, connection strings are usually stored in an XML file named *Web.config*. As the name implies, you can use a *Web.config* file in the root of your website to store the site's configuration information, including any connection strings that your site might require. An example of a connection string in a *Web.config* file might look like the following:
+> In WebMatrix, connection strings are usually stored in an XML file named *Web.config*. As the name implies, you can use a *Web.config* file in the root of your website to store the site's configuration information, including any connection strings that your site might require. An example of a connection string in a *Web.config* file might look like the following, where `$CREDENTIAL_PLACEHOLDER$ ...
```
same

wadepickett replied:
```suggestion
> In WebMatrix, connection strings are usually stored in an XML file named *Web.config*. As the name implies, you can use a *Web.config* file in the root of your website to store the site's configuration information, including any connection strings that your site might require. An example of a connection string in a *Web.config* file might look like the following. Note `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key/value pair:
```

wadepickett commented on aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-cs.md:
```diff
@@ -54,11 +54,11 @@ Next, specify the various database connection information (see Figure 2). When y
 **Figure 2**: Specify the Database Connection Information ([Click to view full-size image](configuring-the-production-web-application-to-use-the-production-database-cs/_static/image6.jpg))
 
-The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the  ...
+The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the  ...
```
```suggestion
The production environment database should now be listed in the Server Explorer. Select the database from the Server Explorer and go to the Properties window. There you will find a property named Connection String with the database s connection string. Assuming you are using a Microsoft SQL Server database on production and the SqlClient provider your connection string should look similar to the following. `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair:
```

Rick-Anderson commented on aspnet/web-forms/overview/data-access/advanced-data-access-scenarios/protecting-connection-strings-and-other-configuration-information-cs.md:
```diff
@@ -164,9 +164,9 @@ These tutorials have used Windows authentication. You can tell what authenticati
 The Integrated Security=True and lack of a username and password indicate that Windows authentication is being used. In some connection strings the term Trusted Connection=Yes or Integrated Security=SSPI is used instead of Integrated Security=True, but all three indicate the use of Windows authentication.
 
-The following example shows a connection string that uses SQL authentication. Note the credentials embedded within the connection string:
+The following example shows a connection string that uses SQL authentication. The `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair. Note the credentials are embedded within the connection string:
```
```suggestion
The following example shows a connection string that uses SQL authentication. `$CREDENTIAL_PLACEHOLDER$` is a placeholder for the password key-value pair. Note the credentials are embedded within the connection string:
```

Rick-Anderson commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/choosing-the-right-approach-to-web-deployment.md:
```diff
@@ -91,7 +91,7 @@ In the case of non-administrator users, the Web Management Service (WMSvc) will
 [!code-console[Main](choosing-the-right-approach-to-web-deployment/samples/sample6.cmd)]
 
-For example, suppose a build process is configured to automatically deploy a web application to a staging environment after every successful build. If you used the remote agent approach, you'd need to make the build process identity an administrator on your destination servers. In contrast, using the Web Deploy Handler approach you can give a non-administrator user&#x2014;**FABRIKAM\stagingdeploy ...
+For example, suppose a build process is configured to automatically deploy a web application to a staging environment after every successful build. If you used the remote agent approach, you'd need to make the build process identity an administrator on your destination servers. In contrast, using the Web Deploy Handler approach you can give a non-administrator user&#x2014;**FABRIKAM\stagingdeploy ...
```
```suggestion
For example, suppose a build process is configured to automatically deploy a web application to a staging environment after every successful build. If you used the remote agent approach, you'd need to make the build process identity an administrator on your destination servers. In contrast, using the Web Deploy Handler approach you can give a non-administrator user&#x2014;**FABRIKAM\stagingdeployer** in this case&#x2014;permission to a specific IIS website only, and the build process can provide these credentials to deploy the web package. Note the following example is using `$CREDENTIAL_PLACEHOLDER$` as a placeholder for the password key-value pair.
```

Rick-Anderson commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/choosing-the-right-approach-to-web-deployment/samples/sample7.cmd:
```diff
@@ -3,7 +3,7 @@ msdeploy.exe
         computerName='https://STAGEWEB1:8172/MSDeploy.axd?site=DemoSite',
         userName='FABRIKAM\stagingdeployer',
-        password='Pa$$w0rd',
+        password=$CREDENTIAL_PLACEHOLDER$,
         authtype='Basic', 
```
No, I thought I asked you to replace this with NTLM so you can remove the password?

Rick-Anderson commented on aspnet/web-forms/overview/getting-started/getting-started-with-aspnet-45-web-forms/membership-and-administration/samples/sample3.cs:
```diff
@@ -41,7 +42,7 @@ internal void AddUserAndRole()
         Email = "canEditUser@wingtiptoys.com"
       };
-      IdUserResult = userMgr.Create(appUser, "Pa$$word1");
+      IdUserResult = userMgr.Create(appUser, ConfigurationManager.AppSettings["AppUserPasswordKey"]);
```
Does that work in Web Forms?

wadepickett replied:
I had tested in a Web Forms app, putting in the web.config and pulling back in through the code this way which worked.  It was problematic trying to test it on the _original_ app sample, it doesn't exist anymore.  There are just some files left it refers to. The original app code example does not exist, instead it points off to samples site, and it is not there.  If there is a more proper way to do this for web forms though let me know.

Rick-Anderson commented on aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-cs.md:
```diff
@@ -54,11 +54,11 @@ Next, specify the various database connection information (see Figure 2). When y
 
 **Figure 2**: Specify the Database Connection Information ([Click to view full-size image](configuring-the-production-web-application-to-use-the-production-database-cs/_static/image6.jpg))
```
https://docs.microsoft.com/en-us/aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-cs has too many passwords to salvage. I think we should delete this file and redirect to one of the ASP.NET Core/SQL Server to Azure tutorials.

Rick-Anderson replied:
Get the traffic on this. I don't think many folks are reading the entire doc.

wadepickett replied:
I sent the data by email to you.  You are correct, the few hitting the doc are bouncing out.

wadepickett replied:
will delete.

Rick-Anderson commented on aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-cs.md:
```diff
@@ -197,7 +197,7 @@ By default, the keys used for encryption and validation are generated automatica
 
 When working in a web farm setting or sharing authentication tickets across applications on the same server, you will need to configure the &lt;machineKey&gt; element in the affected applications so that their decryptionKey and validationKey values match up.
```
https://docs.microsoft.com/en-us/aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-cs

This also has too many passwords to salvage. I'm afraid this one needs to be deleted too.
Please review all the doc's that are flagged and search for "password". If they have more than 2 occurrences, let me know.

wadepickett replied:
-vb version has the same issues.  Will remove that one was well.

wadepickett commented on aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-vb/samples/sample5.xml:
```diff
@@ -12,10 +12,10 @@
  decryptionKey="1513F567EE75F7FB5AC0AC4D79E1D9F25430E3E2F1BCDD3370BCFC4EFC97A541"
 
- validationKey="32CBA563F26041EE5B5FE9581076C40618DCC1218F5F447634EDE8624508A129"
+ validationKey=$CREDENTIAL_PLACEHOLDER$
```
```suggestion
 validationKey="32CBA563F26041EE5B5FE9581076C40618DCC1218F5F447634EDE8624508A129"
```

wadepickett commented on aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-cs.md:
```diff
@@ -197,7 +197,7 @@ By default, the keys used for encryption and validation are generated automatica
 When working in a web farm setting or sharing authentication tickets across applications on the same server, you will need to configure the &lt;machineKey&gt; element in the affected applications so that their decryptionKey and validationKey values match up.
 
-While neither of the above scenarios applies to our sample application, we can still specify explicit decryptionKey and validationKey values and define the algorithms to be used. Add a &lt;machineKey&gt; setting to the Web.config file:
+While neither of the above scenarios applies to our sample application, we can still specify explicit decryptionKey and validationKey values and define the algorithms to be used. Add a &lt;machineKey&gt; setting to the Web.config file. Note the following example is using `$CREDENTIAL_PLACEHOLDER$` as placeholder for the `decryptionKey` and `validationKey` key-value pairs:
```
```suggestion
While neither of the above scenarios applies to our sample application, we can still specify explicit decryptionKey and validationKey values and define the algorithms to be used. Add a &lt;machineKey&gt; setting to the Web.config file:
```

wadepickett commented:
Removed anything related to the following tutorials which instead will be deleted,  deleted in TOC + forwarded to best topic available:

- [Configuring the Production Web Application to Use the Production Database (C#)](https://docs.microsoft.com/en-us/aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-cs)
- [Configuring the Production Web Application to Use the Production Database (VB)](https://docs.microsoft.com/en-us/aspnet/web-forms/overview/older-versions-getting-started/deploying-web-site-projects/configuring-the-production-web-application-to-use-the-production-database-vb)
- [Forms Authentication Configuration and Advanced Topics (C#)](https://docs.microsoft.com/en-us/aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-cs)
- [Forms Authentication Configuration and Advanced Topics (VB)](https://docs.microsoft.com/en-us/aspnet/web-forms/overview/older-versions-security/introduction/forms-authentication-configuration-and-advanced-topics-vb)

Rick-Anderson commented on aspnet/web-forms/overview/deployment/configuring-server-environments-for-web-deployment/choosing-the-right-approach-to-web-deployment/samples/sample7.cmd:
```diff
@@ -3,8 +3,7 @@ msdeploy.exe
         computerName='https://STAGEWEB1:8172/MSDeploy.axd?site=DemoSite',
         userName='FABRIKAM\stagingdeployer',
-        password='Pa$$w0rd',
         authtype='Basic', 
```
Need the NTLM flag

wadepickett replied:
Just a note on where we ended up on this one.  We decided to back to Basic auth plus the placeholder  $CREDENTIAL_PLACEHOLDER$

Rick-Anderson commented:
> Removed anything related to the following tutorials which instead will be deleted, deleted in TOC + forwarded to best topic available:
> 
not seeing the removed files or redirects.

wadepickett commented:
Sorry, I meant to add in there the deleted topics , TOC updates and redirects are going into a new PR (so this one can go out right now)   
I corrected my comment above: "Removed anything related to the following tutorials which instead will be moved to separate PR, deleted,  deleted in TOC + forwarded to best topic available:  
"

wadepickett commented:
@Rick-Anderson does this one look OK to merge?     I am working on a new PR with just the deleted topics, toc changes/redirects at the moment which I will have in by tonight.  I am keeping those separate so we can move this portion.  Thanks for the tremendous help.

Rick-Anderson commented:
@wadepickett  thanks for taking care of this.
Can you close out the internal issues this addresses?
